runtime/plaid/src/apis/cryptography/aes.rs

```diff
@@ -77,41 +71,42 @@ impl Cryptography {
         params: &str,
         module: Arc<PlaidModule>,
     ) -> Result<String, ApiError> {
-        if self.aes.is_none() {
-            return Err(ApiError::CryptographyError(
+        if let Some(aes) = &self.aes {
+            let payload: AesEncryptPayload = serde_json::from_str(&params)
+                .map_err(|_| ApiError::CryptographyError("Failed to parse payload".to_string()))?;
+
+            if !self.can_module_perform_aes_action(
+                &module.name,
+                &payload.key_id,
+                AesAction::Encrypt,
+            ) {
+                return Err(ApiError::CryptographyError(
+                    "Missing key or operation not permitted".to_string(),
+                ));
+            }
+
+            info!(
+                "Performing an AES encryption with local key [{}] on behalf of module [{module}]",
+                payload.key_id
+            );
+
+            let key = aes
+                .key_specs
+                .get(&payload.key_id.to_string())
+                // unwrap OK because we checked above that we have permission to execute this action, so the key must exist
+                .unwrap()
```

Comment:
I would modify the above code to return the `key` iff you have permissions to do the operation. That will cleanly remove this unwrap.